src/main/java/ch/njol/skript/expressions/ExprFurnaceTime.java

```diff
@@ -0,0 +1,309 @@
+							furnace.setCookTimeTotal(0);
+							furnace.update(true);
+						}
+					}
+				} else if (type == FurnaceExpressions.BURNTIME) {
+					if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+						burnEvent.setBurnTime(0);
+					} else {
+						for (Block block : getExpr().getArray(event)) {
+							Furnace furnace = (Furnace) block.getState();
+							furnace.setBurnTime((short) 0);
+							furnace.update(true);
+						}
+					}
+				}
+			}
+			case REMOVE -> {
+				if (type == FurnaceExpressions.COOKTIME) {
+					for (Block block : getExpr().getArray(event)) {
+						Furnace furnace = (Furnace) block.getState();
+						furnace.setCookTime((short) Math.min(furnace.getCookTime() - providedTime, 0));
+						furnace.update(true);
+					}
+				} else if (type == FurnaceExpressions.TOTALCOOKTIME) {
+					if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+						startEvent.setTotalCookTime(Math.min(startEvent.getTotalCookTime() - providedTime, 0));
+					} else {
+						for (Block block : getExpr().getArray(event)) {
+							Furnace furnace = (Furnace) block.getState();
+							furnace.setCookTimeTotal(Math.min(furnace.getCookTimeTotal() - providedTime, 0));
+							furnace.update(true);
+						}
+					}
+				} else if (type == FurnaceExpressions.BURNTIME) {
+					if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+						burnEvent.setBurnTime(Math.min(burnEvent.getBurnTime() - providedTime, 0));
+					} else {
+						for (Block block : getExpr().getArray(event)) {
+							Furnace furnace = (Furnace) block.getState();
+							furnace.setBurnTime((short) Math.min(furnace.getBurnTime() - providedTime, 0));
+							furnace.update(true);
+						}
+					}
+				}
+			}
+			case ADD -> {
+				if (type == FurnaceExpressions.COOKTIME) {
+					for (Block block : getExpr().getArray(event)) {
+						Furnace furnace = (Furnace) block.getState();
+						furnace.setCookTime((short) (furnace.getCookTime() + providedTime));
+						furnace.update(true);
+					}
+				} else if (type == FurnaceExpressions.TOTALCOOKTIME) {
+					if (!explicitlyBlock && event instanceof FurnaceStartSmeltEvent startEvent) {
+						startEvent.setTotalCookTime(startEvent.getTotalCookTime() + providedTime);
+					} else {
+						for (Block block : getExpr().getArray(event)) {
+							Furnace furnace = (Furnace) block.getState();
+							furnace.setCookTimeTotal(furnace.getCookTimeTotal() + providedTime);
+							furnace.update(true);
+						}
+					}
+				} else if (type == FurnaceExpressions.BURNTIME) {
+					if (!explicitlyBlock && event instanceof FurnaceBurnEvent burnEvent) {
+						burnEvent.setBurnTime(burnEvent.getBurnTime() + providedTime);
+					} else {
+						for (Block block : getExpr().getArray(event)) {
+							Furnace furnace = (Furnace) block.getState();
+							furnace.setBurnTime((short) (furnace.getBurnTime() + providedTime));
+							furnace.update(true);
+						}
+					}
+				}
+			}
+		}
+
+	}
```

Comment:
This REALLY needs helper methods to reduce duplicate code.

Reply:
Do you mean by making a protected function for each one that handles its own thing. Like a `changeBurnTime`

Reply:
it's up to you how to structure it. They'd be private, though